### PR TITLE
Remove FontMetrics trait

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: sixtyfps_runtime/rendering_backends/gl/fonts.rs [hunk 1]
```diff
@@ -305,66 +305,30 @@ impl Font {
...
+    text: &str,
+    max_width: Option<f32>,
+) -> Size {
+    let cached_font = item_graphics_cache
+        .get_or_update(graphics_cache, || {
+            Some(super::ItemGraphicsCacheEntry::Font(FONT_CACHE.with(|cache| {
+                cache.borrow_mut().font(
+                    font_request_fn(),
+                    scale_factor.get(),
+                    // FIXME: there is no dependency to the text property
```
Comment (tronical) [inline sixtyfps_runtime/rendering_backends/gl/fonts.rs:322]: Ah, this is I guess what you were referring to as dependency in the chat?

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs [hunk 2]
```diff
@@ -534,26 +533,62 @@ impl PlatformWindow for GraphicsWindow {
...
+        let cache_data = text_input
+            .cached_rendering_data
+            .get_or_update(&self.graphics_cache, || {
+                Some(crate::ItemGraphicsCacheEntry::Font(crate::fonts::FONT_CACHE.with(|cache| {
+                    cache.borrow_mut().font(
+                        text_input
+                            .unresolved_font_request()
+                            .merge(&self.default_font_properties.as_ref().get()),
+                        scale_factor,
+                        &text_input.text(),
```
Comment (tronical) [inline sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:575]: I guess this also needs a FIXME comment like in the other function?

**Review by tronical — APPROVED** [review]

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs (hunk 2 above)
Comment (ogoffart) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:575]: this adds a dependency.  so there is nothing to fix there.

Comment (tronical) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:575]: Ah yes, oops, I misread

**Pull request merged**